pod.go

```diff
@@ -600,6 +596,8 @@ func doFetchPod(podConfig PodConfig) (*Pod, error) {
 		return nil, err
 	}
 
+	agentConfig := newAgentConfig(podConfig)
+
```

Comment:
Why are we creating the `agentConfig` earlier here ?

Reply:
I thought I was cleaning up the sequence but I had not realized that was `agentConfig` we were getting from this. Nevermind, I am gonna move this back !